src/crypto/PSAOperationalKeystore.cpp

```diff
@@ -209,5 +254,53 @@ void PSAOperationalKeystore::ReleasePendingKeypair()
     mIsPendingKeypairActive = false;
 }
 
+CHIP_ERROR PSAOperationalKeystore::MoveOpKeysFromKVSToITS(FabricIndex fabricIndex) const
+{
+    Crypto::SensitiveDataBuffer<TLV::EstimateStructOverhead(sizeof(uint16_t), P256SerializedKeypair::Capacity())> buf;
+    uint16_t keySize = static_cast<uint16_t>(buf.Capacity());
+
+    if (CHIP_NO_ERROR ==
+        mStorage->SyncGetKeyValue(DefaultStorageKeyAllocator::FabricOpKey(fabricIndex).KeyName(), buf.Bytes(), keySize))
+    {
+        PersistentP256Keypair keypair(fabricIndex);
+        // Do not allow overwriting the existing key and just remove it from KVS
+        if (!keypair.Exists())
+        {
+            // Read the operational private and public keys from the TLV entry
+            buf.SetLength(static_cast<size_t>(keySize));
+            TLV::ContiguousBufferTLVReader reader;
+            reader.Init(buf.Bytes(), buf.Length());
+            ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+            TLV::TLVType containerType;
+            ReturnErrorOnFailure(reader.EnterContainer(containerType));
+            ReturnErrorOnFailure(reader.Next(kOpKeyVersionTag));
+            uint16_t opKeyVersion;
+            ReturnErrorOnFailure(reader.Get(opKeyVersion));
+            VerifyOrReturnError(opKeyVersion == kOpKeyVersion, CHIP_ERROR_VERSION_MISMATCH);
+            ReturnErrorOnFailure(reader.Next(kOpKeyDataTag));
+
+            ByteSpan keyData;
+            ReturnErrorOnFailure(reader.GetByteView(keyData));
+            ReturnErrorOnFailure(reader.ExitContainer(containerType));
+            ReturnErrorOnFailure(reader.VerifyEndOfContainer());
+
+            VerifyOrReturnError(keyData.size() == (kP256_PrivateKey_Length + kP256_PublicKey_Length), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+            // Migrate the operational keys
```

Comment:
All code in this method, above, should probably be using public APIs of PersistentStorageOperationalKeystore, rather than partially re-implement  all loading here, and then do the integration work below as well.

In other words:

- Don't reimplement the reading of the keys from persistent storage, rely on PersistentStorageOperationalKeystore.
- You can keep the code in this block in the method, as it is PSA-specific